sickbeard/manual_search.py

```diff
@@ -189,52 +190,65 @@ def get_provider_cache_results(indexer, show_all_results=None, perform_search=No
                           quality, release_group, version, seeders, leechers, size, time \
-                          FROM '{provider_id}' WHERE indexerid = ?".format(provider_id=cur_provider.get_id())
-            additional_sql = " AND episodes LIKE ? AND season = ?"
+                          FROM '{provider_id}' WHERE indexerid = ? AND quality > 0 ".format(provider_id=cur_provider.get_id())
+            additional_sql = " AND episodes LIKE ? AND season = ? "
+
+            # The params are always the same for both queries
+            add_params = [cur_provider.provider_type.title(), cur_provider.image_name(),
+                          cur_provider.name, cur_provider.get_id(), minseed, minleech, show]
 
+            # If were not looking for all results, meaning don't do the filter on season + ep, add sql
             if not int(show_all_results):
-                sql_return = main_db_con.select(common_sql + additional_sql,
-                                                (cur_provider.provider_type.title(), cur_provider.image_name(),
-                                                 cur_provider.name, cur_provider.get_id(),
-                                                 minseed, minleech, show, "%|{0}|%".format(sql_episode), season))
-            else:
-                sql_return = main_db_con.select(common_sql,
-                                                (cur_provider.provider_type.title(), cur_provider.image_name(),
-                                                 cur_provider.name, cur_provider.get_id(), minseed, minleech, show))
-
-        if sql_return:
-            for item in sql_return:
-                found_items.append(dict(item))
-
-            # Store the last table update, we'll need this to compare later
-            provider_results['last_prov_updates'][cur_provider.get_id()] = str(sql_return[0]['lastupdate'])
-        else:
-            provider_results['last_prov_updates'][cur_provider.get_id()] = "0"
+                common_sql += additional_sql
+                add_params += ["%|{0}|%".format(sql_episode), season]
+
+            # Add the created sql, to lists, that are used down below to perform one big UNIONED query
+            combined_sql_q.append(common_sql)
+            combined_sql_params += add_params
+
+            # Get the last updated cache items timestamp
+            last_update = main_db_con.select(b"select max(time) as lastupdate from '{provider_id}'".format(provider_id=cur_provider.get_id()))
+            provider_results['last_prov_updates'][cur_provider.get_id()] = last_update[0]['lastupdate'] if last_update[0]['lastupdate'] else 0
+
+    # Check if we have the combined sql strings
+    if combined_sql_q:
+        sql_prepend = b"SELECT * FROM ("
+        sql_append = b") ORDER BY CASE quality WHEN '{quality_unknown}' THEN -1 ELSE CAST(quality as DECIMAL) END DESC, " \
+                     b" seeders DESC, isproper DESC".format(quality_unknown=Quality.UNKNOWN)
```

Comment:
@p0psicles 
shouldn't be `quality desc, proper desc, seeders desc` ?

my old PR: `key=lambda k: (try_int(k['quality']), try_int(k['is_proper']), try_int(k['seeders']))`

@duramato